nevergrad/parametrization/core.py

```diff
@@ -0,0 +1,497 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import uuid
+import warnings
+from collections import OrderedDict
+import typing as t
+import numpy as np
+
+
+BP = t.TypeVar("BP", bound="BaseParameter")
+P = t.TypeVar("P", bound="Parameter")
+D = t.TypeVar("D", bound="Dict")
+
+
+class Descriptors(t.NamedTuple):
+    """Provides access to a set of descriptors for the parametrization
+    This can be used within optimizers.
+    """
+    deterministic: bool = True
+    continuous: bool = True
```

Comment:
I'm not sure if I understand what a continuous parameter is

Reply:
basically, the parametrization acts as a function between the optimizer and the actual function. Knowing whether the whole instrumentation+function is continuous can be useful for the optimizer.
A few examples:
- instrumentation of a scalar is continuous
- instrumentation of an integer is not continuous
- instrumentation of Ordered discrete is not continiuous
- instrumentation of Softmax categorical is continuous (!) but noisy (we sample based on softmax values, that evolve continuously, but the sampling acts like a noise)
- instrumentation of Softmax categorical in deterministic mode (just sample the one with max probability) is not continuous but not noisy

Reply:
So having 2 parameters (1 scalar + 1 integer) will not behave the same as a single Dict parameter which contains 1 scalar and 1 integer?

Reply:
what do you mean by "2 parameters"?

Reply:
I'm just wondering how this will be used in practice I guess, and how it makes sense to aggregate those to the higher level

Reply:
oh, knowing that the whole (function+parametrization) is continuous or not, deterministic or not,  can be used inside the optimizers to choose a better strategy. Olivier is currently working on competence maps which basically does exactly that: depending on the case, it selects different algorithms

Reply:
Right makes sense thanks!